File: application/src/components/DbControls/src/com/eas/dbcontrols/CellRenderEvent.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package com.eas.dbcontrols;

import com.bearsoft.gui.grid.data.CellData;
import com.bearsoft.rowset.Row;
import com.eas.client.events.PublishedSourcedEvent;
import com.eas.script.HasPublished;
import com.eas.script.NoPublisherException;
import com.eas.script.ScriptFunction;
import jdk.nashorn.api.scripting.JSObject;

/**
 *
 * @author mg
 */
public class CellRenderEvent extends PublishedSourcedEvent {

    protected Object id;
    protected Object columnId;
    protected CellData cell;
    protected Row object;

    public CellRenderEvent(HasPublished source, Object id, Object columnId, CellData cell, Row object) {
        super(source);
        this.columnId = columnId;
        this.id = id;
        this.cell = cell;
        this.object = object;
    }

    @ScriptFunction(jsDoc = ""
            + "/**\n"
            + " * The primary key of the data object.\n"
            + " */")
    public Object getId() {
        return id;
    }

    @ScriptFunction(jsDoc = ""
            + "/**\n"
            + " * The cell's column ID.\n"
            + " */")
    public Object getColumnId() {
        return columnId;
    }

    @ScriptFunction(jsDoc = ""
            + "/**\n"
            + " * The \"abstract\" cell.\n"
            + " */")
    public CellData getCell() {
        return cell;
    }

    @ScriptFunction(jsDoc = ""
            + "/**\n"
            + " * The cell's row object.\n"
            + " */")
    public Row getObject() {
        return object;
    }

    @Override
    public Object getPublished() {
        if (published == null) {
            if (publisher == null || !publisher.isFunction()) {
                throw new NoPublisherException();
            }
            published = publisher.call(null, new Object[]{this});
        }
        return published;
    }

    private static JSObject publisher;

    public static void setPublisher(JSObject aPublisher) {
        publisher = aPublisher;
    }

}

File: application/api/db-controls/cell-render-event.js
(function() {
    var javaClass = Java.type("com.eas.dbcontrols.CellRenderEvent");
    javaClass.setPublisher(function(aDelegate) {
        return new P.CellRenderEvent(aDelegate);
    });
    
    /**
     * Generated constructor.
     * @constructor CellRenderEvent CellRenderEvent
     */
    P.CellRenderEvent = function () {
        var maxArgs = 0;
        var delegate = arguments.length > maxArgs ?
              arguments[maxArgs] 
            : new javaClass();

        Object.defineProperty(this, "unwrap", {
            value: function() {
                return delegate;
            }
        });
        if(P.CellRenderEvent.superclass)
            P.CellRenderEvent.superclass.constructor.apply(this, arguments);
        delegate.setPublished(this);
        Object.defineProperty(this, "columnId", {
            get: function() {
                var value = delegate.columnId;
                return P.boxAsJs(value);
            }
        });
        if(!P.CellRenderEvent){
            /**
             * The cell's column ID.
             * @property columnId
             * @memberOf CellRenderEvent
             */
            P.CellRenderEvent.prototype.columnId = {};
        }
        Object.defineProperty(this, "id", {
            get: function() {
                var value = delegate.id;
                return P.boxAsJs(value);
            }
        });
        if(!P.CellRenderEvent){
            /**
             * The primary key of the data object.
             * @property id
             * @memberOf CellRenderEvent
             */
            P.CellRenderEvent.prototype.id = {};
        }
        Object.defineProperty(this, "source", {
            get: function() {
                var value = delegate.source;
                return P.boxAsJs(value);
            }
        });
        if(!P.CellRenderEvent){
            /**
             * The source object of the event.
             * @property source
             * @memberOf CellRenderEvent
             */
            P.CellRenderEvent.prototype.source = {};
        }
        Object.defineProperty(this, "cell", {
            get: function() {
                var value = delegate.cell;
                return P.boxAsJs(value);
            }
        });
        if(!P.CellRenderEvent){
            /**
             * The "abstract" cell.
             * @property cell
             * @memberOf CellRenderEvent
             */
            P.CellRenderEvent.prototype.cell = {};
        }
        Object.defineProperty(this, "object", {
            get: function() {
                var value = delegate.object;
                return P.boxAsJs(value);
            }
        });
        if(!P.CellRenderEvent){
            /**
             * The cell's row object.
             * @property object
             * @memberOf CellRenderEvent
             */
            P.CellRenderEvent.prototype.object = {};
        }
    };
})();